plugins/exploreVideo/components/searchBar.jsx

```diff
@@ -4,11 +4,28 @@ const propTypes = require('prop-types');
 const React = require('react');
 
 class SearchBar extends React.Component {
+  static getQS() {
+    return (typeof URL === 'undefined') ? '' : new URL(window.location.href).search;
+  }
+
+  // output sample '49.25,-123.1' or ''
+  static getGeoCode(qs) {
+    const matches = /(-?\d+(\.\d+)?),\s*(-?\d+(\.\d+)?)/.exec(qs);
+    return (matches) ? matches[0] : '';
+  }
+
+  static get defaults() {
+    return {
+      geocode: '13.7524000,100.5021833',
+      instruction: 'Keyword or GeoCode'
+    };
+  }
+
```

Comment:
Refactor so functions may be exercised by unit tests and defaults may be read to ensure expected rendered result